### ISSUE TITLE
Calculation of cross-entropy

### ISSUE THREAD

**Issue by kuatroka** [martin-gorner/tensorflow-mnist-tutorial#31]
Hi Martin,
In mnist_1.0_softmax.py you have these lines to describe and calculate cross-entropy:

log takes the log of each element, * multiplies the tensors element by element
reduce_mean will add all the components in the tensor
so here we end up with the total cross-entropy for all images in the batch
`cross_entropy = -tf.reduce_mean(Y_ * tf.log(Y)) * 1000.0  # normalized for batches of 100 images,`
*10 because  "mean" included an unwanted division by 10

Could you please explain a bit more what are these numbers mean?
1. Why are we multiplying the final value by 1000.0 if the batch is of size 100? I thought maybe it was because of 1000 iterations, but in the code we have 2000 iterations.
2. "*10  because  "mean" included an unwanted division by 10" - what does this phrase relate to, I don't see any multiplication by 10.
Thanks

**Comment by kakawait** [comment]
I think

```
1000.0 = BATCH_SIZE * 10
```

with 

> *10 because "mean" included an unwanted division by 10 

Moreover (I'm not an expert, someone can confirm back) but multiplying with `BATCH_SIZE` is just a normalization without any consequences thus can be removed.

Futhermore that way to compute _cross entropy_ is more pedagogic else `TF` is offering a built-in function (that avoid any problem with _unwanted division_) https://github.com/martin-gorner/tensorflow-mnist-tutorial/blob/master/mnist_3.0_convolutional.py#L83

**Comment by martin-gorner** [comment]
The magnitude of the cross entropy does not matter at all for training.
This adjustment is just to make training and test cross-entropy use the same scale so that I can meaningfully display them on the same graph.

The shape of the predictions vector Y is [batch_size, 10] so reduce_mean gives me cross_ent_per_image/10. That is why I multiply is by 10*1000, to obtain the cross-entropy per 100 images, because that looks good on the graphs.
